modules/service/src/main/scala/lucuma/odb/graphql/snippet/ObservationSnippet.scala

```diff
@@ -104,44 +151,113 @@ object ObservationSnippet {
         ObjectMapping(
           tpe = ObservationType,
           fieldMappings = List(
-            SqlField("id", ObservationTable.Id, key = true),
-            SqlField("programId", ObservationTable.ProgramId, hidden=true),
-            SqlField("existence", ObservationTable.Existence, hidden = true),
-            SqlField("name", ObservationTable.Name),
-            SqlField("status", ObservationTable.Status),
-            SqlField("activeStatus", ObservationTable.ActiveStatus),
-            SqlObject("program", Join(ObservationTable.ProgramId, ProgramTable.Id)),
+            SqlField("id", ObservationView.Id, key = true),
+            SqlField("programId", ObservationView.ProgramId, hidden=true),
+            SqlField("existence", ObservationView.Existence, hidden = true),
+            SqlField("name", ObservationView.Name),
+            SqlField("status", ObservationView.Status),
+            SqlField("activeStatus", ObservationView.ActiveStatus),
+            SqlObject("constraintSet"),
```

Comment:
I'll admit I don't fully understand this part.  Does it find the `constraintSet` field in the GraphQL schema and match it up with the `ConstraintSetType` object mapping below?